File: dart/scripts/hello_modules.dart
void main(List<String> args) {
    var parsed = parse_args(args);
    var controller = HelloController();
    controller.say(parsed['world'] ?? '');
}

class HelloController {
    String greeting = 'Hello';

    void say(String to) {
        print('$greeting, $to!');
    }
}

Map<String, String> parse_args(List<String> args) {
    return {
        'world': args[0],
    };
}

File: swift/scripts/HelloModules.swift
import Foundation

class HelloController {
  var greeting = "Hello"

  func say(to: String) {
    print("\(self.greeting), \(to)!`")
  }
}

func parseArgs() -> [String:String]? {
  guard let world = CommandLine.arguments.dropFirst().first else { return nil }
  return [ "world": world ]
}

func main() {
  guard let args = parseArgs() else { return }
  let controller = HelloController()
  if let world = args["world"] {
    controller.say(to: world)
  }
}

main()
